File: Problem_of_the_Day/PreorderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void compute(TreeNode* root, vector<int> &result){
        if(root==NULL){
            return;
        }
          result.push_back(root->val);
          compute(root->left,result);
          compute(root->right,result);
    }
    vector<int> preorderTraversal(TreeNode* root) {
        //preorder: root left right
        vector<int> result;
        compute(root,result);
        return result;
    }
};

File: Tree/PreorderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def compute(self, root, result):
        if root:
            result.append(root.val)
            self.compute(root.left,result)
            self.compute(root.right,result)
            
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        self.compute(root,result)
        return result
